config/config.go

```diff
@@ -841,6 +846,24 @@ func (c *Config) Valid() error {
 		}
 	}
 
+	// check capacity and the quota when OOMUseTmpStorage is enabled
+	if c.OOMUseTmpStorage {
+		if c.TempStorageQuota == 0 {
+			// Print warning
+		} else {
+			var stat syscall.Statfs_t
+			err := syscall.Statfs(c.TempStoragePath, &stat)
+			if err != nil {
+				return err
+			}
+			capacityBytes := stat.Bavail * uint64(stat.Bsize)
+			quotaBytes := uint64(c.TempStorageQuota * 1024)
+			if quotaBytes > capacityBytes {
+				return fmt.Errorf("value of [temp-storage-quota] exceeds the capacity of the [temp-storage-path] directory")
+			}
```

Comment:
Use [build constraints](https://golang.org/pkg/go/build/#hdr-Build_Constraints) to avoid failing with the WASM platform.